api_server/src/request/boot_source.rs

```diff
@@ -46,4 +48,25 @@ mod tests {
                 receiver
             ))))
     }
+
+    #[test]
+    fn test_into_parsed_request_different() {
+        let body = BootSourceConfig {
+            kernel_image_path: String::from("/foo/bar"),
+            initrd_path: None,
+            boot_args: Some(String::from("foobar")),
+        };
+        let initrd_body = BootSourceConfig {
+            kernel_image_path: String::from("/foo/bar"),
+            initrd_path: Some(String::from("/bar/baz")),
+            boot_args: Some(String::from("foobar")),
+        };
+        let (sender, receiver) = oneshot::channel();
+        assert!(!body
```

Comment:
This assert doesn't look good. If you look at the previous test, it checks that body.into_parsed_request equals ConfigureBootSource(same_body, sender). In that test body and same_body are the same thing. We created two object with the same data because `BootSourceConfig` does not implement copy so basically we just create two object with same data to test the "equality".

Reply:
Understood, test updated, thanks!